packages/service-worker/src/update.ts

```diff
@@ -50,29 +74,54 @@ export class SwUpdate {
 
   constructor(private sw: NgswCommChannel) {
     if (!sw.isEnabled) {
+      this.versionUpdates = NEVER;
       this.available = NEVER;
       this.activated = NEVER;
       this.unrecoverable = NEVER;
       return;
     }
-    this.available = this.sw.eventsOfType<UpdateAvailableEvent>('UPDATE_AVAILABLE');
+    this.versionUpdates = this.sw.eventsOfType<VersionEvent>(
+        ['VERSION_DETECTED', 'VERSION_INSTALLATION_FAILED', 'VERSION_READY']);
+    this.available = this.versionUpdates.pipe(
+        filter((evt: VersionEvent): evt is VersionReadyEvent => evt.type === 'VERSION_READY'),
+        map(evt => ({
+              type: 'UPDATE_AVAILABLE',
+              current: evt.currentVersion,
+              available: evt.latestVersion,
+            })));
     this.activated = this.sw.eventsOfType<UpdateActivatedEvent>('UPDATE_ACTIVATED');
     this.unrecoverable = this.sw.eventsOfType<UnrecoverableStateEvent>('UNRECOVERABLE_STATE');
   }
 
-  checkForUpdate(): Promise<void> {
+  /**
+   * Checks for an update and waits until the new version is ready for activation.
```

Comment:
Let's be more explicit that this includes downloading the necessary assets from the server:

```suggestion
   * Checks for an update and waits until the new version is downloaded from the server and ready
   * for activation.
```